whylog/config/__init__.py

```diff
@@ -28,6 +35,87 @@ def _filter_rule_set(self, parsers_list):
     def _get_locations_for_logs(self, logs_types_list):
         pass
 
+    def add_rule(self, user_rule_intent):
+        rule = self._create_rule_from_user_rule_intent(user_rule_intent)
+        self._save_rule_definition(rule.get_rule_in_form_to_save())
+        self._save_parsers_definition(rule.get_rule_parsers_in_form_to_save())
+
+    def _save_rule_definition(self, rule_definition):
+        with open(self._rules_path, "a") as rules_file:
+            rules_file.write(yaml.safe_dump(rule_definition, explicit_start=True))
+
+    def _save_parsers_definition(self, parser_definitions):
+        with open(self._parsers_path, "a") as parsers_file:
+            parsers_file.write(yaml.safe_dump_all(parser_definitions, explicit_start=True))
+
+    def _create_rule_from_user_rule_intent(self, user_rule_intent):
+        parsers_dict = self._create_parsers_from_parsers_intents(user_rule_intent)
+        effect = parsers_dict.get(user_rule_intent.effect_id)
+        parsers_dict.pop(user_rule_intent.effect_id)
+        causes, parser_ids_mapper = self._create_causes_list_with_clue_index(
+            parsers_dict, user_rule_intent
+        )
+        constraints = self._create_constraints_list(parser_ids_mapper, user_rule_intent)
+        return Rule(causes, effect, constraints)
+
+    def _create_parsers_from_parsers_intents(self, user_rule_intent):
+        return dict(
+            (
+                intent_id, RegexParser(
+                    str(uuid.uuid4()), parser_intent.regex, parser_intent.primary_key_groups,
```

Comment:
move this to classmethod `RegexParserFactory.create_from_intent(parser_intent)`. Otherwise `YamlConfig` private method will be responsible for how `RegexParser` objects are created. It would not be right.